shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -485,6 +503,8 @@ void setTextInputEditingState(View view, TextInputChannel.TextEditState state) {
     if (mRestartInputPending) {
       mImm.restartInput(view);
       mRestartInputPending = false;
+    } else {
+      mEditable.clearBatchDeltas();
```

Comment:
Nit: Consider commenting here and other places where `clearBatchDeltas` is called to explain why it's being called there.  At a high level it makes sense to clear it when the framework is sending an update, but it seems really nuanced to know when exactly to clear.  If it's easily explainable then a comment might be good.